src/cascadia/TerminalApp/ShortcutActionDispatch.h

```diff
@@ -22,51 +24,53 @@ namespace winrt::TerminalApp::implementation
-        TYPED_EVENT(CloseOtherTabs,       TerminalApp::ShortcutActionDispatch, Microsoft::Terminal::Settings::Model::ActionEventArgs);
-        TYPED_EVENT(CloseTabsAfter,       TerminalApp::ShortcutActionDispatch, Microsoft::Terminal::Settings::Model::ActionEventArgs);
-        TYPED_EVENT(TabSearch,            TerminalApp::ShortcutActionDispatch, Microsoft::Terminal::Settings::Model::ActionEventArgs);
-        TYPED_EVENT(MoveTab,              TerminalApp::ShortcutActionDispatch, Microsoft::Terminal::Settings::Model::ActionEventArgs);
-        TYPED_EVENT(BreakIntoDebugger,    TerminalApp::ShortcutActionDispatch, Microsoft::Terminal::Settings::Model::ActionEventArgs);
-        TYPED_EVENT(FindMatch,            TerminalApp::ShortcutActionDispatch, Microsoft::Terminal::Settings::Model::ActionEventArgs);
-        TYPED_EVENT(TogglePaneReadOnly,   TerminalApp::ShortcutActionDispatch, Microsoft::Terminal::Settings::Model::ActionEventArgs);
-        TYPED_EVENT(NewWindow,            TerminalApp::ShortcutActionDispatch, Microsoft::Terminal::Settings::Model::ActionEventArgs);
+        DECLARE_ACTION(CopyText);
+        DECLARE_ACTION(PasteText);
+        DECLARE_ACTION(OpenNewTabDropdown);
+        DECLARE_ACTION(DuplicateTab);
+        DECLARE_ACTION(NewTab);
+        DECLARE_ACTION(CloseWindow);
+        DECLARE_ACTION(CloseTab);
+        DECLARE_ACTION(ClosePane);
+        DECLARE_ACTION(SwitchToTab);
+        DECLARE_ACTION(NextTab);
+        DECLARE_ACTION(PrevTab);
+        DECLARE_ACTION(SendInput);
+        DECLARE_ACTION(SplitPane);
+        DECLARE_ACTION(TogglePaneZoom);
+        DECLARE_ACTION(AdjustFontSize);
+        DECLARE_ACTION(ResetFontSize);
+        DECLARE_ACTION(ScrollUp);
+        DECLARE_ACTION(ScrollDown);
+        DECLARE_ACTION(ScrollUpPage);
+        DECLARE_ACTION(ScrollDownPage);
+        DECLARE_ACTION(ScrollToTop);
+        DECLARE_ACTION(ScrollToBottom);
+        DECLARE_ACTION(OpenSettings);
+        DECLARE_ACTION(ResizePane);
+        DECLARE_ACTION(Find);
+        DECLARE_ACTION(MoveFocus);
+        DECLARE_ACTION(ToggleShaderEffects);
+        DECLARE_ACTION(ToggleFocusMode);
+        DECLARE_ACTION(ToggleFullscreen);
+        DECLARE_ACTION(ToggleAlwaysOnTop);
+        DECLARE_ACTION(ToggleCommandPalette);
+        DECLARE_ACTION(SetColorScheme);
+        DECLARE_ACTION(SetTabColor);
+        DECLARE_ACTION(OpenTabColorPicker);
+        DECLARE_ACTION(RenameTab);
+        DECLARE_ACTION(OpenTabRenamer);
+        DECLARE_ACTION(ExecuteCommandline);
+        DECLARE_ACTION(CloseOtherTabs);
+        DECLARE_ACTION(CloseTabsAfter);
+        DECLARE_ACTION(TabSearch);
+        DECLARE_ACTION(MoveTab);
+        DECLARE_ACTION(BreakIntoDebugger);
+        DECLARE_ACTION(FindMatch);
+        DECLARE_ACTION(TogglePaneReadOnly);
+        DECLARE_ACTION(NewWindow);
+        DECLARE_ACTION(IdentifyWindow);
+        DECLARE_ACTION(IdentifyWindows);

```

Comment:
i hate to say it, but this seems like it just straight-up belongs in the xmacros PR. I totally get it though.

Reply:
Yea, it's kinda a half measure right now. A taste of goodness to come - a carrot for more reviews 😛